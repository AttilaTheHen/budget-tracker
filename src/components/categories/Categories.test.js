import React from 'react';
import { shallow } from 'enzyme';
import Categories from './Categories';
import toJSON from 'enzyme-to-json';

describe('Categories', () => {
  it('renders add if no category prop', () => {
    const categories = [
      { key: '123', timestamp: 'new Date()', name: 'transportation', budget: 400 },
      { key: '456', timestamp: 'new Date()', name: 'other', budget: 200 },
      { key: '789', timestamp: 'new Date()', name: 'more food', budget: 500 },
    ];

    const wrapper = shallow(<Categories
      categories={categories}
      onUpdate={jest.fn()}
    />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});