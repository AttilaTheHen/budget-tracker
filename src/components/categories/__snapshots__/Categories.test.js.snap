// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories renders categories 1`] = `
<ul
  className="categories"
>
  <Connect(Category)
    category={
      Object {
        "budget": 400,
        "key": "123",
        "name": "transportation",
        "timestamp": "new Date()",
      }
    }
    key="123"
  />
  <Connect(Category)
    category={
      Object {
        "budget": 200,
        "key": "456",
        "name": "other",
        "timestamp": "new Date()",
      }
    }
    key="456"
  />
  <Connect(Category)
    category={
      Object {
        "budget": 500,
        "key": "789",
        "name": "more food",
        "timestamp": "new Date()",
      }
    }
    key="789"
  />
</ul>
`;
